Compatibility with ccache

pip-installing on Python 3.7 results in an error *"could not configure cfitsio 3370patch"*, is it a known issue ? (I could not find a similar issue, so I'm opening this one).

```
Collecting fitsio
  Using cached https://files.pythonhosted.org/packages/9c/cb/f52534b71f4d99916723af2994898904015b9a1bf0286a165182d0374bbf/fitsio-0.9.11.tar.gz
Requirement already satisfied: numpy in /home/simon/.pyenv/versions/3.7.0/lib/python3.7/site-packages (from fitsio) (1.15.4)
Building wheels for collected packages: fitsio
  Running setup.py bdist_wheel for fitsio ... error
  Complete output from command /home/simon/.pyenv/versions/3.7.0/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/simon/pip-install-reb18mp4/fitsio/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/simon/pip-wheel-aym8pbhk --python-tag cp37:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/fitsio
  copying fitsio/util.py -> build/lib.linux-x86_64-3.7/fitsio
  copying fitsio/test.py -> build/lib.linux-x86_64-3.7/fitsio
  copying fitsio/fitslib.py -> build/lib.linux-x86_64-3.7/fitsio
  copying fitsio/__init__.py -> build/lib.linux-x86_64-3.7/fitsio
  running build_ext
  replacing '-O3' with '-O2' to address gcc bug
  checking build system type... x86_64-unknown-linux-gnu
  checking host system type... x86_64-unknown-linux-gnu
  checking target system type... x86_64-unknown-linux-gnu
  checking for gcc... ccache
  checking whether the C compiler works... yes
  checking for C compiler default output file name... a.out
  checking for suffix of executables... configure: error: in `/tmp/simon/pip-install-reb18mp4/fitsio/build/temp.linux-x86_64-3.7/cfitsio3370patch':
  configure: error: cannot compute suffix of executables: cannot compile and link
  See `config.log' for more details
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/simon/pip-install-reb18mp4/fitsio/setup.py", line 203, in <module>
      "build_ext": build_ext_subclass,
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 188, in run
      self.run_command('build')
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/home/simon/.pyenv/versions/3.7.0/lib/python3.7/distutils/command/build_ext.py", line 339, in run
      self.build_extensions()
    File "/tmp/simon/pip-install-reb18mp4/fitsio/setup.py", line 77, in build_extensions
      RANLIB=self.compiler.ranlib,
    File "/tmp/simon/pip-install-reb18mp4/fitsio/setup.py", line 159, in configure_cfitsio
      raise ValueError("could not configure cfitsio %s" % self.cfitsio_version)
  ValueError: could not configure cfitsio 3370patch
```

It looks configure is having trouble building the C code. See this line:

> configure: error: cannot compute suffix of executables: cannot compile and link
  See `config.log' for more details

I don't think this error is specific to python 3.7.

--- a/setup.py
+++ b/setup.py
@@ -81,8 +81,11 @@ class build_ext_subclass(build_ext):
                     print("replacing '-O3' with '-O2' to address "
                           "gcc bug")
                     val='-O2'
-                CC.append(val) 
-                    
+                if val=='ccache':
+                    print("removing ccache from the compiler options")
+                    continue
+                CC.append(val)
+
             self.configure_cfitsio(
                 CC=CC, 
                 ARCHIVE=self.compiler.archiver, 

